lib/castle/commands/impersonate.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module Castle
+  module Commands
+    # builder for impersonate command
+    class Impersonate
+      def initialize(context)
+        @context = context
+      end
+
+      def build(options = {})
+        validate!(options)
+        context = ContextMerger.call(@context, options[:context])
+        context = ContextSanitizer.call(context)
+
+        Castle::Command.new('impersonate',
+                            options.merge(context: context,
+                                          sent_at: Castle::Utils::Timestamp.call),
+                            :post)
+      end
+
+      private
+
+      def validate!(options)
+        %i[user_id].each do |key|
```

Comment:
From impersonate endpoint:
```
requires :user_id, type: String
requires :user_agent, type: String
requires :ip, type: String
optional :impersonator, type: String
optional :client_id, type: String
```
Please update validation
```

Reply:
@lluft - I'm changing the endpoint . It will look like this before merging this:  ```
        requires :user_id, type: String
        optional :impersonator, type: String, desc: 'Impersonator email'
        optional :context, type: Hash, desc: 'Request context. (includes Ip, User-Agent)'
        optional :timestamp, desc: 'Timestamp of the event', types: [String, Time]
        optional :sent_at, type: String
        optional :received_at, type: String
```